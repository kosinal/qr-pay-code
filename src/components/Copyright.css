.copyright {
  position: fixed;
  bottom: 0.5rem;
  left: 0.5rem;
  font-size: 0.75rem;
  color: var(--text-primary);
  opacity: 0.7;
  z-index: 1000;
}

.copyright:hover {
  opacity: 1;
}

.copyright a {
  color: var(--text-primary);
  text-decoration: none;
}

.copyright a:hover {
  text-decoration: underline;
}

/* Mobile responsive layout */
@media (max-width: 768px) {
  .copyright {
    position: static;
    text-align: center;
    padding: 1rem;
    margin-top: 0;
    display: block;
    width: 100%;
  }
}
