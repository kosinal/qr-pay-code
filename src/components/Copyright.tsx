import React from 'react';
import packageJson from '../../package.json';
import './Copyright.css';

export const Copyright: React.FC = () => {
  return (
    <div className="copyright">
      © {new Date().getFullYear()} Lukas Kosina · v{packageJson.version}
    </div>
  );
};
