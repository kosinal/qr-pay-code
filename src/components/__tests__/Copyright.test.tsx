import { render, screen } from '@testing-library/react';
import { describe, it, expect, vi } from 'vitest';
import { Copyright } from '../Copyright';
import packageJson from '../../../package.json';

describe('Copyright Component', () => {
  it('renders copyright symbol and author name', () => {
    render(<Copyright />);
    expect(screen.getByText(/© .* Lukas Kosina/)).toBeInTheDocument();
  });

  it('displays current year', () => {
    const currentYear = new Date().getFullYear();
    render(<Copyright />);
    expect(screen.getByText(new RegExp(`© ${currentYear}`))).toBeInTheDocument();
  });

  it('displays version from package.json', () => {
    render(<Copyright />);
    expect(screen.getByText(new RegExp(`v${packageJson.version}`))).toBeInTheDocument();
  });

  it('displays correct format with all elements', () => {
    const currentYear = new Date().getFullYear();
    render(<Copyright />);
    expect(
      screen.getByText(`© ${currentYear} Lukas Kosina · v${packageJson.version}`)
    ).toBeInTheDocument();
  });

  it('applies copyright className', () => {
    const { container } = render(<Copyright />);
    expect(container.querySelector('.copyright')).toBeInTheDocument();
  });

  it('renders as a div element', () => {
    const { container } = render(<Copyright />);
    const copyrightDiv = container.querySelector('.copyright');
    expect(copyrightDiv?.tagName).toBe('DIV');
  });

  it('uses current year from Date object', () => {
    vi.useFakeTimers();
    vi.setSystemTime(new Date('2024-06-15'));

    const { rerender } = render(<Copyright />);
    expect(screen.getByText(/© 2024/)).toBeInTheDocument();

    vi.setSystemTime(new Date('2025-03-20'));
    rerender(<Copyright />);
    expect(screen.getByText(/© 2025/)).toBeInTheDocument();

    vi.useRealTimers();
  });

  it('contains middot separator between author and version', () => {
    render(<Copyright />);
    expect(screen.getByText(/·/)).toBeInTheDocument();
  });

  it('renders version number with v prefix', () => {
    render(<Copyright />);
    const text = screen.getByText(/v\d+\.\d+\.\d+/);
    expect(text).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { container } = render(<Copyright />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
