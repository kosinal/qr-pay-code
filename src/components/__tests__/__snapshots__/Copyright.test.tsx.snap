// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Copyright Component > matches snapshot 1`] = `
<div
  class="copyright"
>
  © 
  2025
   Lukas Kosina · v
  1.0.3
</div>
`;
